providers/os/id/clouddetect/clouddetect.go

```diff
@@ -29,16 +29,25 @@ var detectors = []detectorFunc{
 	gcp.Detect,
 }
 
+const AssetKind = "virtualmachine"
+
 type detectResult struct {
 	platformId         string
 	platformName       string
 	relatedPlatformIds []string
 }
 
-func Detect(conn shared.Connection, p *inventory.Platform) (PlatformID, PlatformName, []RelatedPlatformID) {
+type PlatformInfo struct {
+	ID                 string
+	Name               string
+	Kind               string
+	RelatedPlatformIDs []string
+}
+
+func Detect(conn shared.Connection, p *inventory.Platform) PlatformInfo {
 	mgr, err := smbios.ResolveManager(conn, p)
 	if err != nil {
-		return "", "", nil
+		return PlatformInfo{"", "", "", nil}
```

Comment:
Should this return nil? But we need to check the error handling on the other side.